server/src/services/address.service.ts

```diff
@@ -0,0 +1,98 @@
+import { Transaction } from 'sequelize/types';
+import User from '../models/user.model';
+import UserAddress from '../models/userAddress.model';
+
+import { ICreateAddress, IUpdateAddress } from '../interfaces/addressService.interface';
+
+class AddressService {
+  private async _updateUserDefaultAddressByUserId(
+    transaction: Transaction,
+    user_id: string,
+    address_id: number | null
+  ) {
+    const [userUpdateSuccess] = await User.update(
+      { default_address: address_id },
+      {
+        where: {
+          id: user_id,
+        },
+        transaction,
+      }
+    );
+    if (!userUpdateSuccess) {
+      throw Error(); // unexpected error
+    }
+  }
+
+  public async createAddress(transaction: Transaction, user_id: string, address: ICreateAddress): Promise<UserAddress> {
+    const isDefaultAddress: boolean | undefined = address.default_address;
+    const newUserAddress: UserAddress = (await UserAddress.create({ ...address, user_id }, { transaction })).get();
+
+    if (isDefaultAddress) {
+      await this._updateUserDefaultAddressByUserId(transaction, user_id, newUserAddress.id);
+    }
+    return newUserAddress;
+  }
+
+  public async getAddressesByUserId(user_id: string): Promise<UserAddress[]> {
+    const results: UserAddress[] = await UserAddress.findAll({
+      where: {
+        user_id,
+      },
+      attributes: ['id', 'name', 'address', 'phone', 'zip_code'],
+    });
+
+    return results;
+  }
+
+  public async updateAddressById(transaction: Transaction, user_id: string, address: IUpdateAddress): Promise<void> {
+    // check this address_id is owner's default_address
+    const isDefaultAddress =
+      (await User.findOne({ where: { id: user_id, default_address: address.id } })) !== null ? true : false;
+
+    const default_address: boolean | undefined = address.default_address;
+    const [updateSuccess] = await UserAddress.update(
+      { ...address },
+      {
+        where: {
+          id: address.id,
+          user_id,
+        },
+        transaction,
+      }
+    );
+    if (!updateSuccess) {
+      // address id not match
+      throw new Error(); // unexpected Error
+    }
+
+    if (default_address) {
+      await this._updateUserDefaultAddressByUserId(transaction, user_id, address.id);
+    } else if (isDefaultAddress) {
+      await this._updateUserDefaultAddressByUserId(transaction, user_id, null);
+    }
+  }
+
+  public async deleteAddressById(transaction: Transaction, user_id: string, id: number): Promise<void> {
+    const isDefaultAddress =
+      (await User.findOne({ where: { id: user_id, default_address: id } })) !== null ? true : false;
```

Comment:
```suggestion
    const isDefaultAddress =
      !!(await User.findOne({ where: { id: user_id, default_address: id } }));
```
이렇게 표현할 수도 있긴 한데...
한줄로 표현하기엔 긴 수식같은 느낌이 조금은 드네요 🤔

위에 같은 수식이 있기도 하고,
아예 함수로 분리해서 쓰는 편은 어떠려나요?
무조건적으로 짧은 것보단 조금만 가독성이 개선되면 좋을 것 같습니다! ☺️